src/forklift/slack.py

```diff
@@ -0,0 +1,362 @@
+            if server_status['success']:
+                success = ':white_check_mark:'
+
+            message.add(SectionBlock(f'{success} *{server_status["name"]}*'))
+
+            if server_status.get('has_service_issues', False):
+                items = split(server_status['problem_services'], MAX_CONTEXT_ELEMENTS)
+
+                for item in items:
+                    message.add(ContextBlock(item))
+            else:
+                message.add(ContextBlock([':rocket: All services started']))
+
+            if len(server_status['message']) > 0:
+                message.add(ContextBlock([server_status['message']]))
+
+            message.add(SectionBlock('Datasets shipped'))
+
+            shipped_data = ['No data updated']
+            if len(server_status['successful_copies']) > 0:
+                shipped_data = server_status['successful_copies']
+
+            items = split(shipped_data, MAX_CONTEXT_ELEMENTS)
+
+            for item in items:
+                message.add(ContextBlock(item))
+
+            message.add(DividerBlock())
+
+    message.add(SectionBlock('*Pallets Report*'))
+
+    for pallet in report['pallets']:
+        success = ':fire:'
+
+        if pallet['success']:
+            success = ':heavy_check_mark:'
+
+        message.add(SectionBlock(f'{success} *{pallet["name"].split(":")[2]}*'))
+
+        post_copy_processed = shipped = ':red_circle:'
+        if pallet['post_copy_processed']:
+            post_copy_processed = ':white_check_mark:'
+        if pallet['shipped']:
+            shipped = ':white_check_mark:'
+
+        elements = [pallet['total_processing_time']]
+
+        if pallet['message']:
+            elements.append(pallet['message'])
+
+        elements.append(f'Post copy processed: {post_copy_processed}')
+        elements.append(f'Shipped: {shipped}')
+
+        items = split(elements, MAX_CONTEXT_ELEMENTS)
+        for item in items:
+            message.add(ContextBlock(item))
+
+    return message.get_messages()
+
+
+class BlockType(Enum):
+    '''available block type enums
+    '''
+    SECTION = 'section'
+    DIVIDER = 'divider'
+    CONTEXT = 'context'
+
+
+class Block(ABC):
+    '''Basis block containing attributes and behavior common to all
+    Block is an abstract class and cannot be sent directly.
+    '''
+
+    def __init__(self, block):
+        self.type = block
+
+    def _attributes(self):
+        return {'type': self.type.value}
+
+    @abstractmethod
+    def _resolve(self):
+        pass
+
+    def __repr__(self):
+        return dumps(self._resolve(), indent=2)
+
+
+class Text():
+    '''A text class formatted using slacks markdown syntax
+    '''
+
+    def __init__(self, text):
+        self.text = text
+
+    def _resolve(self):
+        text = {
+            'type': 'mrkdwn',
+            'text': self.text,
+        }
+
+        return text
+
+    @staticmethod
+    def to_text(text, max_length=None):
+        if max_length and len(text) > max_length:
```

Comment:
```suggestion
        if max_length and len(str(text)) > max_length:
```

Would this fix the latest error? From what I can deduce, the issue is that the database pallet message is an ExecuteError object and this code expects it to be a string. Which is what the rest of forklift expects as well. I took a look at the pallet and I can't understand how it's happening since it appears that it sets it as a string as well. But either way, I wonder if this would take care of the issue.

Reply:
shouldn't hurt!